src/util/TickUtils.ts

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash';
+
+import { TickFormatter } from '../cartesian/CartesianAxis';
+import { getAngledRectangleWidth } from './CartesianUtils';
+import { getStringSize } from './DOMUtils';
+import { getEveryNthWithCondition } from './getEveryNthWithCondition';
+import { Size, CartesianViewBox, CartesianTickItem } from './types';
+
+export function getAngledTickWidth(contentSize: Size, unitSize: Size, angle: number) {
+  const size = { width: contentSize.width + unitSize.width, height: contentSize.height + unitSize.height };
+
+  return getAngledRectangleWidth(size, angle);
+}
+
+export function getInitialStartAndEnd(viewBox: CartesianViewBox, sign: number, sizeKey: string) {
+  let start, end;
+
+  const { x, y, width, height } = viewBox;
+  if (sign === 1) {
+    start = sizeKey === 'width' ? x : y;
+    end = sizeKey === 'width' ? x + width : y + height;
+  } else {
+    start = sizeKey === 'width' ? x + width : y + height;
+    end = sizeKey === 'width' ? x : y;
```

Comment:
nit: can we move the check `sizeKey === 'width'` to a variable?

Reply:
Good idea, done.